quilt/tools/store.py

```diff
@@ -141,14 +177,38 @@ def ls_packages(self):
         """
         List packages in this store.
         """
-        packages = [
-            (user, pkg[:-len(self.PACKAGE_FILE_EXT)])
-            for user in os.listdir(self._path)
-            if os.path.isdir(os.path.join(self._path, user))
-            for pkg in os.listdir(os.path.join(self._path, user))
-            if pkg.endswith(self.PACKAGE_FILE_EXT)]
+        packages = []
+        pkgdir = os.path.join(self._path, self.PKG_DIR)
+        for user in os.listdir(pkgdir):
+            for pkg in os.listdir(os.path.join(pkgdir, user)):
+                pkgpath = os.path.join(pkgdir, user, pkg)           
+                pkgmap = {h : [] for h in os.listdir(os.path.join(pkgpath, Package.CONTENTS_DIR))}
+                for tag in os.listdir(os.path.join(pkgpath, Package.TAGS_DIR)):
```

Comment:
Just realized something: on Windows and OS X, tags and versions will be case-insensitive. That may be ok - but we should think about it.

Reply:
Good point. Case insensitive matching seems fine to me. Ideally we'd have consistent behavior on all systems though.